lib/Actions.js

```diff
@@ -149,9 +245,46 @@ Actions.prototype = {
     },
 
     /**
+     * Wait for an element is on the page and visible
+     * @param {string} selector
+     * @param {number} [timeout]
+     * @param {number} [interval]
+     * @returns {Promise}
+     */
+    waitElementIsVisible: function (selector, timeout, interval) {
+        debug('._waitElementIsVisible() ' + selector);
+        return this.wait(/* @covignore */ function (selector) {
+            var nodes = Array.prototype.slice.call(Sizzle(selector), 0);
+            return nodes.some(function (node) {
+                return node.offsetWidth !== 0 && node.offsetHeight !== 0;
```

Comment:
You can't understand because it's a client side )) http://stackoverflow.com/questions/14122013/detect-if-an-element-is-visible-without-using-jquery